teps/0029-step-workspaces.md

```diff
@@ -57,6 +71,69 @@ that has access to those contents as well.
 - A Task author can still use the volume "hack" to attach `Workspaces` to `Sidecars` in
 combination with the feature proposed here.
 
+## Proposal
+
+### Add `workspaces` to `Steps`
+
+1. Add a `workspaces` list to `Steps`.
+2. Allow `workspaces` from the `Task` to be explicitly named in that list like this:
+
+    ```yaml
+    workspaces:
+    - name: my-sensitive-workspace
+    steps:
+    - name: foo
+      workspaces:
+      - name: my-sensitive-workspace
+    ```
+3. When a `workspace` is listed in a Step, it is no longer automatically mounted - either to
```

Comment:
apologies in advance for coming late to this, and possibly re-introducing prior discussion, as well as coming in without hands on experience with sidecars

but the `is no longer automatically mounted` text implies a change in behaviour and "breaking API change" of sorts

is my non-SME interpretation correct?  if so, should this TEP be explicit in stating "a breaking API change is being suggested to facilitate X/Y/Z" ?

some of the referenced design docs may claim that more decisively (apologies, I did not search them for such), but I would contend that detail of such ilk should be surfaced in the TEP.

conversely, if this is not a breaking API sort of change, perhaps some clarification somewhere in the TEP as to why that is the case would help those coming in later on to the discussion

WDYT @sbwsg ?

thanks

Reply:
@gabemontero thanks for reviewing!

"is no longer automatically mounted" -> Sorry for the confusion.  It's definitely not a breaking API change because we haven't had "Step-level" workspaces before this.  So it's a new behaviour that the user opts in to by adding the workspace to a Step.  They couldn't do this before - add a workspace to a Step - and by doing so they're effectively saying "I only want this Step to have access to this workspace."

Here's what a Task with workspaces currently looks like:

```yaml
spec:
  workspaces:
  - name: foo # All Steps get this workspace
  steps:
  - image: ubuntu
    script: # ...
```

and here's the new thing we're adding:

```yaml
spec:
  workspaces:
  - name: foo
  steps:
  - image: ubuntu
    workspaces:
    - name: foo # Only this Step gets this workspace
    script: # ...
```

Reply:
I'll take another breeze through this doc and try to clarify that language.

Reply:
ah OK good to know @sbwsg 

the before / after yaml's together like that are concise and make the net change clear ... adding the before yaml right before the existing after yaml and changing that text would benefit me at least 

thanks

Reply:
Added an explicit before/after example at the top of the `Proposal` section. Cheers!

Reply:
OK  the new adding new API / behavior but preserving existing API / behavior verbiage makes it very clear now @sbwsg thanks!

LGTM

I'll circle back tomorrow to see if others have chimed in or our planning to chime in soon for the "multi-company" sign off, otherwise I can do it if needed.

Reply:
@vdemeester says he is going to look at this today @sbwsg .... I'll defer to him on LGTM-ing